src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -3580,6 +3580,14 @@ ExtractInsertPartitionKeyValue(Query *query)
 		/* single-row INSERT with a constant partition column value */
 		singlePartitionValueConst = (Const *) targetExpression;
 	}
+	else if (IsA(targetExpression, ArrayExpr))
+	{
+		singlePartitionValueConst =
+			(Const *) evaluate_expr((Expr *) targetExpression,
```

Comment:
This feels a bit hazardous, since the array could contain arbitrary expressions that maybe should not be evaluated here.

Could we check that the array is constant and construct the appropriate Const ourselves?